hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfspp/tests/configuration_test.h

```diff
@@ -114,105 +111,6 @@ void writeDamagedConfig(const std::string& filename, Args... args) {
   out.open(filename);
   out << stream.rdbuf();
 }
-
-// TempDir: is deleted on destruction
-class TempFile {
```

Comment:
Other than reducing the size of this file, is there any benefit of having this? Or this is just a refactor at this point?

Reply:
As I've mentioned in the corresponding JIRA - https://issues.apache.org/jira/browse/HDFS-16174, header files should contain only the declaration of the class and method signatures. The implementation must be done in the cc files. Otherwise, these classes get recompiled everytime they're **#include**d, leading to longer compilation times. Also, it's not a good C++ practice to combine declaration and implementation in the header file, unless it's a template class or method.

Another thing that I've done in this PR is to make this into a _test utility library_, so that it's more modular and can be imported anywhere in libhdfs++.

Reply:
Those are valuable but let's make this a refactor PR to make it more explicit.